core/src/test/java/org/kiwix/kiwixmobile/core/dao/HistoryDaoTest.kt

```diff
@@ -0,0 +1,171 @@
+        .getHistoryList(false, "")
+        .map(HistoryItem::historyTitle)
+
+      assertTrue(historyTitleList.contains(historyItem1.historyTitle))
+      assertTrue(historyTitleList.contains(historyItem2.historyTitle))
+      assertTrue(historyTitleList.contains(historyItem3.historyTitle))
+    }
+  }
+
+  @Nested
+  inner class DeleteHistoryDesigns {
+    @Test
+    fun `deleteHistory should delete selected item`() {
+      val historyItem1 = mockkHistoryItemWithTitle("1")
+      val historyItem2 = mockkHistoryItemWithTitle("2")
+      val historyItem3 = mockkHistoryItemWithTitle("3")
+      historyDao.saveHistory(historyItem1)
+      historyDao.saveHistory(historyItem2)
+      historyDao.saveHistory(historyItem3)
+      every { historyItem1.databaseId } returns 1
+      historyDao.deleteHistory(listOf(historyItem1))
+
+      val historyTitleList: List<String> = historyDao
+        .getHistoryList(false, "")
+        .map(HistoryItem::historyTitle)
+
+      assertFalse(historyTitleList.contains(historyItem1.historyTitle))
+      assertTrue(historyTitleList.contains(historyItem2.historyTitle))
+      assertTrue(historyTitleList.contains(historyItem3.historyTitle))
+    }
+
+    @Test
+    fun `deleteHistory should delete multiple selected items`() {
+      val historyItem1 = mockkHistoryItemWithTitle("1")
+      val historyItem2 = mockkHistoryItemWithTitle("2")
+      val historyItem3 = mockkHistoryItemWithTitle("3")
+      historyDao.saveHistory(historyItem1)
+      historyDao.saveHistory(historyItem2)
+      historyDao.saveHistory(historyItem3)
+      every { historyItem1.databaseId } returns 1
+      every { historyItem2.databaseId } returns 2
+      historyDao.deleteHistory(listOf(historyItem1, historyItem2))
+
+      val historyTitleList: List<String> = historyDao
+        .getHistoryList(false, "")
+        .map(HistoryItem::historyTitle)
+
+      assertFalse(historyTitleList.contains(historyItem1.historyTitle))
+      assertFalse(historyTitleList.contains(historyItem2.historyTitle))
+      assertTrue(historyTitleList.contains(historyItem3.historyTitle))
+    }
+  }
+
+  @Test
+  fun `deleteAllHistory should delete all history`() {
+    val historyItem1 = mockkHistoryItemWithTitle("1")
+    val historyItem2 = mockkHistoryItemWithTitle("2")
+    val historyItem3 = mockkHistoryItemWithTitle("3")
+    historyDao.saveHistory(historyItem1)
+    historyDao.saveHistory(historyItem2)
+    historyDao.saveHistory(historyItem3)
+    historyDao.deleteAllHistory()
+
+    val historyTitleList: List<String> = historyDao
+      .getHistoryList(false, "")
+      .map(HistoryItem::historyTitle)
+
+    assertFalse(historyTitleList.contains(historyItem1.historyTitle))
+    assertFalse(historyTitleList.contains(historyItem2.historyTitle))
+    assertFalse(historyTitleList.contains(historyItem3.historyTitle))
+  }
+}
+
+fun mockkHistoryItemWithTitle(historyTitle: String): HistoryListItem.HistoryItem {
+  val historyItem = mockk<HistoryItem>()
```

Comment:
as mentioned elsewhere on this review, it is preferred to just create an instance of a data class, see TestModelFunctions. That class is a bit difficult to use unfortunately because AS doesn't offer great support for sharing classes between test/androidTest and across modules

Reply:
@macgills The only solution I have found so far of changing the databaseID of the pre-inserted historyItem is by mocking the HistoryItem, then using it in the constructor of a HistoryEntity which in turn is used in the constructor of a new HistoryItem. Like this: 

```
val historyItem = HistoryItem(HistoryEntity(mockkHistoryItemWithTitle('title')))
```
Is this ok or can you recommend a more elegant solution? 

The block in TestModelFunctions will look like this: 
```
...
fun historyItem(
  url: String,
  dateString: String,
  databaseId: Long,
  zimId: String,
  zimName: String,
  zimFilePath: String,
  favicon: String?,
  historyTitle: String,
  timeStamp: Long
): HistoryItem =
  createHistoryItemFromHistoryItem(
    mockHistoryItem(
      url,
      dateString,
      databaseId,
      zimId,
      zimName,
      zimFilePath,
      favicon,
      historyTitle,
      timeStamp
    )
  )

private fun createHistoryItemFromHistoryItem(historyItem: HistoryItem) =
  HistoryItem(HistoryEntity(historyItem))

private fun mockHistoryItem(
  url: String,
  dateString: String,
  databaseId: Long,
  zimId: String,
  zimName: String,
  zimFilePath: String,
  favicon: String?,
  historyTitle: String,
  timeStamp: Long
): HistoryItem {
  val mockedHistoryItem = mockk<HistoryItem>()
  every { mockedHistoryItem.historyUrl } returns url
  every { mockedHistoryItem.dateString } returns dateString
  every { mockedHistoryItem.databaseId } returns databaseId
  every { mockedHistoryItem.zimId } returns zimId
  every { mockedHistoryItem.zimName } returns zimName
  every { mockedHistoryItem.zimFilePath } returns zimFilePath
  every { mockedHistoryItem.favicon } returns favicon
  every { mockedHistoryItem.historyTitle } returns historyTitle
  every { mockedHistoryItem.timeStamp } returns timeStamp
  return mockedHistoryItem
}
...
```